### DISCUSSION TITLE
Is there a hook that checks if route is active?

### DISCUSSION THREAD

**Discussion by nklhtv** [TanStack/router#1906] [Q&A] [answered]
The way i currently handle this is like:
```
function useIsRouteActive(options: UseLinkPropsOptions) {
  const linkProps = useLinkProps(options) as any; // eslint-disable-line @typescript-eslint/no-explicit-any
  return linkProps["data-status"] === "active";
}
```
But it requires a cast to any, so i was wondering if there is a better way. Maybe we could extract [this code](https://github.com/TanStack/router/blob/main/packages/react-router/src/link.tsx#L600-L629) into a separate hook and expose it as a public api?

**Comment by schiller-manuel** [comment, marked as answer]
I think this should be possible with `useMatch` once https://github.com/TanStack/router/pull/1738 lands, right?

**Reply by nklhtv** [reply]
Nice, looks like exactly what i need.

**Reply by schiller-manuel** [reply]
FYI this is released already
